RuntimeError: DataLoader worker (pid 4024) is killed by signal: Killed.

No matter how small my train/val set is I always get a runtime error.  Is there a reason for this issue?


Full Stack Trace:
```
root@d88132b833a3:/opt/data/code# rastervision run local tiny_spacenet.py
python -m rastervision.pipeline.cli run_command /opt/data/output/pipeline-config.json analyze
Running analyze command...
2021-01-14 20:49:58:rastervision.core.rv_pipeline.rv_pipeline: INFO - Running analyzers: StatsAnalyzer...
python -m rastervision.pipeline.cli run_command /opt/data/output/pipeline-config.json chip
Running chip command...
2021-01-14 20:50:14:rastervision.core.rv_pipeline.rv_pipeline: INFO - Making train chips for scene: NCCABA011011OrthoSectorTile_scene
2021-01-14 20:50:15:rastervision.core.rv_pipeline.rv_pipeline: INFO - Making train chips for scene: NCCABA011012OrthoSectorTile_scene
2021-01-14 20:50:17:rastervision.core.rv_pipeline.rv_pipeline: INFO - Making validation chips for scene: NCCABA011014OrthoSectorTile_scene
2021-01-14 20:50:19:rastervision.core.rv_pipeline.rv_pipeline: INFO - Making validation chips for scene: NCCABA011015OrthoSectorTile_scene
2021-01-14 20:50:20:rastervision.pipeline.file_system.utils: INFO - Uploading /opt/data/tmp/tmpc0p4cni0/tmpjzacxc_3/output.zip to /opt/data/output/chip/fc4e6074-ffb1-41e1-bc57-583e0c940006.zip
python -m rastervision.pipeline.cli run_command /opt/data/output/pipeline-config.json train
Running train command...
Downloading: "https://download.pytorch.org/models/resnet50-19c8e357.pth" to /opt/data/torch-cache/checkpoints/resnet50-19c8e357.pth
100%|█████████████████████████████████████| 97.8M/97.8M [00:08<00:00, 11.6MB/s]
2021-01-14 20:50:33:rastervision.pytorch_learner.learner: INFO - model=SemanticSegmentationModelConfig(backbone=<Backbone.resnet50: 'resnet50'>, pretrained=True, init_weights=None, type_hint='semantic_segmentation_model') solver=SolverConfig(lr=0.0001, num_epochs=1, test_num_epochs=2, test_batch_sz=4, overfit_num_steps=1, sync_interval=1, batch_sz=2, one_cycle=True, multi_stage=[], type_hint='solver') data=SemanticSegmentationDataConfig(uri='/opt/data/output/chip', train_sz=None, group_uris=None, group_train_sz=None, data_format=<SemanticSegmentationDataFormat.default: 'default'>, class_names=['building', 'background', 'null'], class_colors=['red', 'black', 'black'], img_sz=300, num_workers=4, augmentors=['RandomRotate90', 'HorizontalFlip', 'VerticalFlip'], type_hint='semantic_segmentation_data') predict_mode=False test_mode=False overfit_mode=False eval_train=False save_model_bundle=True log_tensorboard=True run_tensorboard=False output_uri='/opt/data/output/train' type_hint='semantic_segmentation_learner'
2021-01-14 20:55:06:rastervision.pytorch_learner.learner: INFO - train_ds: 14390 items
2021-01-14 20:55:06:rastervision.pytorch_learner.learner: INFO - valid_ds: 10540 items
2021-01-14 20:55:06:rastervision.pytorch_learner.learner: INFO - test_ds: 10540 items
2021-01-14 20:55:09:rastervision.pytorch_learner.learner: INFO - epoch: 0
Training  [------------------------------------]    0%  0d 20:58:36
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/cli.py", line 233, in <module>
    main()
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/cli.py", line 225, in run_command
    runner=runner)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/cli.py", line 202, in _run_command
    command_fn()
  File "/opt/src/rastervision_core/rastervision/core/rv_pipeline/rv_pipeline.py", line 114, in train
    backend.train()
  File "/opt/src/rastervision_pytorch_backend/rastervision/pytorch_backend/pytorch_learner_backend.py", line 69, in train
    learner.main()
  File "/opt/src/rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py", line 152, in main
    self.train()
  File "/opt/src/rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py", line 862, in train
    train_metrics = self.train_epoch()
  File "/opt/src/rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py", line 793, in train_epoch
    output = self.train_step(batch, batch_ind)
  File "/opt/src/rastervision_pytorch_learner/rastervision/pytorch_learner/semantic_segmentation_learner.py", line 99, in train_step
    out = self.post_forward(self.model(x))
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/segmentation/_utils.py", line 22, in forward
    x = self.classifier(x)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/segmentation/deeplabv3.py", line 91, in forward
    res.append(conv(x))
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 547, in __call__
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 343, in forward
    return self.conv2d_forward(input, self.weight)
  File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 340, in conv2d_forward
    self.padding, self.dilation, self.groups)
  File "/opt/conda/lib/python3.6/site-packages/torch/utils/data/_utils/signal_handling.py", line 66, in handler
    _error_if_any_worker_fails()
RuntimeError: DataLoader worker (pid 4024) is killed by signal: Killed. 
/opt/data/output/Makefile:12: recipe for target '2' failed
make: *** [2] Error 1
root@d88132b833a3:/opt/data/code# 
```

Code:

```
# flake8: noqa

from os.path import join, splitext
from os import walk
from rastervision.core.rv_pipeline import *
from rastervision.core.backend import *
from rastervision.core.data import *
from rastervision.pytorch_backend import *
from rastervision.pytorch_learner import *

base_path = '/opt/data/input'
path_tifs = "{}/OrthoMosaicTilesTIFFs2010".format(base_path)
path_geojsons = "{}/OrthoMosaicTilesLabels".format(base_path)



def filter_ext(f):
    _, file_extension = splitext(f)
    if file_extension == ".tif":
        return True

def get_ml_data(p):
    for d,p,files in walk(p):
        return files






def get_config(runner):
    root_uri = '/opt/data/output/'
    base_path = '/opt/data/input'
    channel_order = [0, 1, 2]
    class_config = ClassConfig(
        names=['building', 'background'], colors=['red', 'black'])

    def make_scene(scene_id, image_uri, label_uri):
        """
        - StatsTransformer is used to convert uint16 values to uint8.
        - The GeoJSON does not have a class_id property for each geom,
          so it is inferred as 0 (ie. building) because the default_class_id
          is set to 0.
        - The labels are in the form of GeoJSON which needs to be rasterized
          to use as label for semantic segmentation, so we use a RasterizedSource.
        - The rasterizer set the background (as opposed to foreground) pixels
          to 1 because background_class_id is set to 1.
        """
        raster_source = RasterioSourceConfig(
            uris=[image_uri],
            channel_order=channel_order,
            transformers=[StatsTransformerConfig()])
        vector_source = GeoJSONVectorSourceConfig(
            uri=label_uri, default_class_id=0, ignore_crs_field=True)
        label_source = SemanticSegmentationLabelSourceConfig(
            raster_source=RasterizedSourceConfig(
                vector_source=vector_source,
                rasterizer_config=RasterizerConfig(background_class_id=1)))
        return SceneConfig(
            id=scene_id,
            raster_source=raster_source,
            label_source=label_source)





    path_tifs = "{}/OrthoMosaicTilesTIFFs2010".format(base_path)
    path_geojsons = "{}/OrthoMosaicTilesLabels".format(base_path)



    def filter_ext(f):
        _, file_extension = splitext(f)
        if file_extension == ".tif":
            return True

    def get_ml_data(p):
        for d,p,files in walk(p):
            return files

    training_scenes = []
    validation_scenes = []


    filtered_list_tif = list(filter(filter_ext, get_ml_data(path_tifs)))
    list_geojson = get_ml_data(path_geojsons)
    index = 0
    for geojson, tif in zip(list_geojson, filtered_list_tif):

        file_name, file_extension = os.path.splitext(geojson)
        tif_index = filtered_list_tif.index(file_name + ".tif")


        label = geojson
        aerial = filtered_list_tif[tif_index]
        scene = file_name + "_scene"

        if index < 2:
            training_scenes.append(make_scene(scene, join(path_tifs,aerial), join(path_geojsons,label)))
        elif index > 2 and index < 5:
            validation_scenes.append(make_scene(scene, join(path_tifs,aerial), join(path_geojsons,label)))
        index = index + 1








    dataset = DatasetConfig(
        class_config=class_config,
        train_scenes=training_scenes,
        validation_scenes=validation_scenes)

    # Uses the PyTorch backend for the SemanticSegmentation pipeline.
    chip_sz = 300
    backend = PyTorchSemanticSegmentationConfig(
        model=SemanticSegmentationModelConfig(backbone=Backbone.resnet50),
        solver=SolverConfig(lr=1e-4, num_epochs=1, batch_sz=2))
    chip_options = SemanticSegmentationChipOptions(
        window_method=SemanticSegmentationWindowMethod.random_sample,
        chips_per_scene=10)

    return SemanticSegmentationConfig(
        root_uri=root_uri,
        dataset=dataset,
        backend=backend,
        train_chip_sz=chip_sz,
        predict_chip_sz=chip_sz,
        predict_batch_sz=2,
        chip_options=chip_options)
```

--- a/cookiecutter_template/{{cookiecutter.project_name}}/docker/run
+++ b/cookiecutter_template/{{cookiecutter.project_name}}/docker/run
@@ -95,7 +95,7 @@ fi
 
 if [ "${BASH_SOURCE[0]}" = "${0}" ]
 then
-    docker run --shm-size 8G ${RUNTIME} ${NAME} --rm -it \
+    docker run ${RUNTIME} ${NAME} --rm --ipc=host -it \
         -v "${HOME}"/.rastervision:/root/.rastervision \
         -v ${REPO_ROOT}/rastervision_{{cookiecutter.project_name}}:/opt/src/rastervision_{{cookiecutter.project_name}} \
         -v ${DATA_DIR}:/opt/data \

--- a/docker/run
+++ b/docker/run
@@ -107,7 +107,7 @@ fi
 
 if [ "${BASH_SOURCE[0]}" = "${0}" ]
 then
-    docker run ${RUNTIME} ${NAME} --rm -it \
+    docker run ${RUNTIME} ${NAME} --rm --ipc=host -it \
         -v "${HOME}"/.rastervision:/root/.rastervision \
         -v ${REPO_ROOT}/:/opt/src/ \
         -v ${RASTER_VISION_DATA_DIR}:/opt/data \

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -43,6 +43,7 @@ Bug Fixes
 * Update all relevant saved URIs in config before instantiating Pipeline `#993 <https://github.com/azavea/raster-vision/pull/993>`_
 * Pass verbose flag to batch jobs `#988 <https://github.com/azavea/raster-vision/pull/988>`_
 * Fix: Ensure Integer class_id `#990 <https://github.com/azavea/raster-vision/pull/990>`_
+* Use ``--ipc=host`` by default when running the docker container `#1077 <https://github.com/azavea/raster-vision/pull/1077>`_
 
 Raster Vision 0.12
 -------------------

